core/src/main/java/org/elasticsearch/gateway/ReplicaShardAllocator.java

```diff
@@ -116,6 +116,11 @@ public boolean allocateUnassigned(RoutingAllocation allocation) {
                 continue;
             }
 
+            // if we are allocating a replica because of index creation, no need to go and find a copy, there isn't one...
+            if (shard.allocatedPostIndexCreate() == false) {
```

Comment:
Do you think it makes sense to add `&& IndexMetaData.isUsingSharedFilesystem(shardSettings) == false` here also, so that replicas on shared filesystems can still be allocated on the node with the most "up-to-date" view of the data?